tests/test_utils.py

```diff
@@ -220,40 +218,49 @@ def format_jsonapi_request(type_, data):
     return retval
 
 
-def _test_res_sort_included(result, sortFn=lambda x: x['type']):
+def assert_valid_response(data, expected):
     """
-    Given a jsonapi response with included data(i.e. an Order that includes
-    user, address and items), return the same result with the list of `included`
-    sorted using ``sortFn``.
-
-    : param result: jsonapi structure that needs normalization
-    : param sortFn: sorting function called on every ``included`` resource.
-                   default takes the attribute ``type`` to sort the resources.
-    : type sortFn: ``function``
+    Take a flask app response.data and the expected test result, normalize them
+    sorting the `included` and `errors` lists if present, then assert their
+    equality
     """
-    # safety check.
-    if 'included' not in result:
-        return result
-
-    def sort(r):
-        r['included'] = sorted(r['included'], key=sortFn)
-        return r
-    if type(result) is list:
-        result = [sort(r) for r in result]
-    else:
-        result = sort(result)
-
-    return result
-
-
-def _test_res_sort_errors(e):
-    """
-    Returns the list of errors from a validate_input call, sorted by the
-    errors / source / pointer attribute, allowing proper testing.
-    """
-
-    e['errors'] = sorted(e['errors'], key=lambda e: e['source']['pointer'])
-    return e
+    def sort_data_lists(data, attribute, key):
+        """
+        sort a given data structure's attribute (list) using the given key function
+        """
+        try:
+            data[attribute] = sorted(data[attribute], key=key)
+        except (KeyError, TypeError):
```

Comment:
Mi puo' stare bene che in questa riga ci sia un check sul keyerror perche' l'attributo e' opzionale. Il typeerror no.